### PR TITLE
refactor: extract noise/decoherence into HybridSuperQubits.noise module (#17)

### PR DESCRIPTION
**Pull request by joanjcaceres**
Closes the first step of #17: extract decoherence calculations into a standalone, functional API.

## Summary

- New `HybridSuperQubits.noise` module containing pure functions: `t1_capacitive`, `t1_inductive`, `t1_flux_bias_line`, `tphi_1_over_f`, `tphi_CQPS`, plus the generic `t1_from_spectral_density` builder. All take eigenvalues + operator matrix elements + scalar parameters — no qubit object required.
- The corresponding methods on `QubitBase` are now ~10-line wrappers that resolve `esys`/matrix elements from `self` and delegate. Public signatures unchanged → **no user code breaks**.
- New docs page `docs/api/noise.rst` and a quickstart snippet showing the standalone API.
- Version bumped `0.10.9` → `0.11.0` (new public module, additive).

This is Phase A of the architectural change discussed in #17 with @kyrylo-gr — start with noise because it has the cleanest separation from Hamiltonian machinery and is not tied to any published-paper results. Phase B (sweep dispatchers `_get_t1_vs_paramvals` etc.) will land as a follow-up PR.

## Test plan

- [x] `tests/unit/test_noise.py` — 14 tests with synthetic 2- and 4-level inputs (transition signs, reciprocity, matrix-element scaling, Q_cap callable-vs-float equivalence, generic builder matches channel-specific, edge cases for both Tphi branches).
- [x] `tests/integration/test_noise_parity.py` — 7 wrapper-vs-standalone parity tests on real Fluxonium and Ferbo instances. The Ferbo test exercises the 2nd-order Tphi branch (the trickiest delegated path).
- [x] All 21 new tests pass. The 4 pre-existing failures on `main` (`test_potential_includes_berry_contribution`, `test_ferbo_*_standard_params`) are Berry-related regression baselines from commit 0017d2b — verified to be unrelated by re-running them with this branch's changes stashed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: HybridSuperQubits/qubit_base.py [hunk 1]
```diff
@@ -845,7 +729,7 @@ def get_t1_capacitive_vs_paramvals(
 
             def Q_cap_fun(omega):
                 return (
-                    1e6 * (2 * np.pi * 6e9 / np.abs(omega)) ** 0.7
+                    1/3e-5 * (2 * np.pi * 6e9 / np.abs(omega)) ** 0.7
                 )  # Assuming that Ec is in GHz
```
[the remark this answers is not in the document]
Comment (joanjcaceres) [reply HybridSuperQubits/qubit_base.py:733]: Fixed in de09a56 — `_default_Q_cap` in the new `HybridSuperQubits.noise` module now uses `1/3e-5`, the physically correct value already in use by the sweep path. Flagged as a behavior change in the CHANGELOG (capacitive T1 from `qubit.t1_capacitive()` without an explicit `Q_cap` is now ~30x shorter than in v0.10.x).
